torchdiffeq/_impl/adjoint.py

```diff
@@ -182,22 +162,35 @@ def odeint_adjoint(func, y0, t, *, rtol=1e-7, atol=1e-9, method=None, options=No
         adjoint_atol = atol
     if adjoint_method is None:
         adjoint_method = method
+
```

Comment:
It may be worth copying over this check here:
```
if adjoint_method != method and options is not None and adjoint_options is None:
    raise ValueError("If `adjoint_method != method` then we cannot infer `adjoint_options` from `options`. So as "
                     "`options` has been passed then `adjoint_options` must be passed as well.")
```

Reply:
Done.